sklearn/cross_validation.py

```diff
@@ -700,41 +706,90 @@ def __len__(self):
+        elif np.asarray(train_size).dtype.kind == 'i':
+            if train_size >= n:
+                raise ValueError("train_size=%d should be smaller "
+                                 "than the number of samples %d" %
+                                 (train_size, n))
+            test_size = n - train_size
+        else:
+            raise ValueError("Invalid value for train_size: %r" % train_size)
+    else:  # test_size is not None and train_set is not None:
+        if np.array(test_size).dtype.kind == 'f' and \
+           np.array(train_size).dtype.kind == 'f':
+            if test_size >= 1.:
+                raise ValueError('test_size=%f should be smaller '
+                                 'than 1.0 or be an integer' % test_size)
+            if train_size >= 1.:
+                raise ValueError('train_size=%f should be smaller '
+                                 'than 1.0 or be an integer' % train_size)
+            if train_size + test_size > 1.:
                 raise ValueError('The sum of test_size and train_size = %f, '
                                  'should be smaller than 1.0. Reduce '
                                  'test_size and/or train_size.' %
                                  (train_size + test_size))
-        elif np.asarray(train_size).dtype.kind == 'i':
+        elif np.array(test_size).dtype.kind == 'f' and \
+                np.array(train_size).dtype.kind == 'i':
+            raise ValueError(
+                "Type of test_size and train_sizeis not the same, "
+                "test_size: %r, train_size: %r" %
+                (type(test_size), type(train_size)))
+        elif np.array(test_size).dtype.kind == 'i' and \
+                np.array(train_size).dtype.kind == 'f':
+            raise ValueError(
+                "Type of test_size and train_sizeis not the same, "
+                "test_size: %r, train_size: %r" %
+                (type(test_size), type(train_size)))
+        elif np.array(test_size).dtype.kind == 'i' and \
+                np.array(train_size).dtype.kind == 'i':
+            if test_size >= n:
+                raise ValueError('test_size=%d should be smaller '
+                                 'than the number of samples %d' %
+                                 (test_size, n))
             if train_size >= n:
                 raise ValueError("train_size=%d should be smaller "
                                  "than the number of samples %d" %
                                  (train_size, n))
+            if train_size + test_size > n:
+                raise ValueError('The sum of test_size and train_size = %d, '
+                                 'should be smaller than n. Reduce '
+                                 'test_size and/or train_size.' %
+                                 (train_size + test_size))
         else:
-            raise ValueError("Invalid value for train_size: %r" % train_size)
+            if np.array(test_size).dtype.kind != 'f' and \
```

Comment:
I think we hope the error message should as clear as possable.
It's helpfull for beginner to sklearn.